lib/league_stats.rb

```diff
@@ -59,4 +59,44 @@ def worst_offense
       team.team_id == bad_o.to_s
     end.teamname
   end
+
+  def highest_scoring_visitor
+    away_teams = @game_teams_collection.game_teams.find_all do |team|
+      team.hoa == "away"
+    end
+    by_team = away_teams.group_by do |team|
+      team.team_id
+    end
+    average = by_team.transform_values do |games|
+      games.sum do |game|
+        (game.goals.to_f / games.length).round(2)
+      end
+    end
+    highest_average = average.max_by do |team_id, average|
+      average
+    end
+    @teams_collection.teams.find do |team|
+      team.team_id == highest_average.first
+    end.teamname
+  end
+
+  def highest_scoring_home_team
+    home_teams = @game_teams_collection.game_teams.find_all do |team|
+      team.hoa == "home"
+    end
+    by_team = home_teams.group_by do |team|
+      team.team_id
+    end
```

Comment:
Cool choice of group_by method here to gather the team_ids. It's cleaner for sure